### PR TITLE
#341 Manager can edit master / intern roles

### PR DESCRIPTION
**Pull request by vryazanov**
#341 
* Manager can access to Master / Intern accounts
* tests fixed

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#395](https://github.com/timelesslounge/timelessis/issues/395) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#395](https://github.com/timelesslounge/timelessis/issues/395) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/3ed574d6-3a8c-4ab3-83f6-839176cd55d1); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/factories.py [hunk 1]
```diff
@@ -2,9 +2,10 @@
 import factory
 import random
 
-from timeless import DB, companies
+from timeless import DB
```
Comment (vladarefiev) [inline tests/factories.py]: `from timeless.db import DB`?

File: tests/integration/it_manager_privilages_test.py [hunk 2]
```diff
@@ -75,98 +76,38 @@ def test_cant_access_other_company_employees(app, db_session):
...
-        user_status="on",
-        registration_date=datetime.utcnow(),
-        company_id=my_company.id,
-        email="test@test.com", password="bla", role_id=manager_role.id
-    )
-    db_session.add(me)
+    company = factories.CompanyFactory()
+    manager = factories.RoleFactory()
+    me = factories.EmployeeFactory(company=company, role=manager)
+    colleague = factories.EmployeeFactory(company=company, role=manager)
```
Comment (vladarefiev) [inline tests/integration/it_manager_privilages_test.py]: @vryazanov Thanks you

Comment (vladarefiev) [reply tests/integration/it_manager_privilages_test.py]: @paulodamaso Take a look pls

File: tests/integration/it_manager_privilages_test.py [hunk 3]
```diff
@@ -75,98 +76,38 @@ def test_cant_access_other_company_employees(app, db_session):
...
-        pin_code=7777,
-        account_status="on",
-        user_status="on",
-        registration_date=datetime.utcnow(),
-        company_id=my_company.id,
-        email="test@test.com", password="bla", role_id=manager_role.id
-    )
-    db_session.add(me)
+    company = factories.CompanyFactory()
+    manager = factories.RoleFactory()
```
Comment (vladarefiev) [inline tests/integration/it_manager_privilages_test.py]: @vryazanov let's name `manager_role`

File: timeless/access_control/manager_privileges.py [hunk 4]
```diff
@@ -23,10 +23,19 @@ def __employee_access(method=None, *args, **kwargs):
 
 
 def check_employee(employee_id, method, user):
-    if employee_id == user.id and method == Method.READ:
+    if employee_id == user.id:
         return True
-    else:
-        return user.company_id == Employee.query.get(employee_id).company_id
+
+    employee = Employee.query.get(employee_id)
```
Comment (vladarefiev) [inline timeless/access_control/manager_privileges.py]: @vryazanov Could it be `None`?

File: timeless/employees/models.py [hunk 5]
```diff
@@ -28,6 +28,7 @@ class Employee(TimestampsMixin, DB.Model):
     company = DB.relationship("Company", back_populates="employees")
     items = DB.relationship("Item", back_populates="empolyee")
     history = DB.relationship("ItemHistory", back_populates="employee")
+    role = DB.relationship("Role")
```
Comment (vladarefiev) [inline timeless/employees/models.py]: @vryazanov `back_populates` ?

**Comment by vryazanov** [comment]
@vladarefiev all 4 comments you left have been fixed

**Comment by vladarefiev** [comment]
@rultor good to merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 1 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Job [#395](https://github.com/timelesslounge/timelessis/issues/395) is not in the agenda of @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), can't inspect

**Comment by 0crat** [comment]
The job [#395](https://github.com/timelesslounge/timelessis/issues/395) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@vladarefiev  According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Messages in a job [always](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to.

Please correct your [message](https://github.com/timelesslounge/timelessis/pull/395#discussion_r259638962) by indicating an addressee in the beginning.

**Comment by vladarefiev** [comment]
@ypshenychka confirm

**Comment by ypshenychka** [comment]
@vladarefiev thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

**Pull request merged**